src/main/java/com/pk/service/BooksService.java

```diff
@@ -1,22 +1,26 @@
 package com.pk.service;
 
 import com.pk.model.Book;
+import com.pk.writer.BibTexWriter;
 
 import java.time.format.TextStyle;
+import java.util.HashSet;
 import java.util.Locale;
 
 public class BooksService {
 
     /**
      * <p>Returns string made of all book elements for saving to BibTex format.</p>
-     * @param book Book of which string will be returned
-     * @return String in BibTex format
+     * @param book Book of which string will be returned.
+     * @param keys Unique set of BibTex keys.
+     * @return String in BibTex format.
      */
-    public static String bookToBibTexString(Book book) {
+    public static String bookToBibTexString(Book book, HashSet<String> keys) {
```

Comment:
HashSet zmień na Set

Lepiej jest mieć bardziej ogólny typ. Gdybyśmy w przyszłości zechcieli używać innego typu Set np TreeSet to będzie to wiele prostsze.